Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts

```diff
@@ -1,57 +1,77 @@
-import { QuestionType } from "../../questions/enum-questiontype"
-import { Category } from "../../questions/category.model"
+import { AddCategoryDialogComponent } from './add-category-dialog.component';
+import { DeleteCategoryDialogComponent } from './delete-category-dialog.component';
+import { DeleteQuestionDialogComponent } from './delete-question-dialog.component';
+import { QuestionsService } from '../questions.service';
+import { CategoryService } from '../categories.service';
+import { Question } from '../../questions/question.model';
+import { DifficultyLevel } from '../../questions/enum-difficultylevel';
+import { QuestionType } from '../../questions/enum-questiontype';
+import { Category } from '../../questions/category.model';
+import { RenameCategoryDialogComponent } from './rename-category-dialog.component';
+
 @Component({
     moduleId: module.id,
-    selector: "questions-dashboard",
-    templateUrl: "questions-dashboard.html"
+    selector: 'questions-dashboard',
+    templateUrl: 'questions-dashboard.html'
 })
-export class QuestionsDashboardComponent {
+
+export class QuestionsDashboardComponent implements OnInit {
+    private category: Category = new Category();
+
     questionDisplay: Question[] = new Array<Question>();
     categoryArray: Category[] = new Array<Category>();
-    //To enable enum difficultylevel in template
+    // to enable enum difficultylevel in template
     DifficultyLevel = DifficultyLevel;
-    //To enable enum questiontype in template 
+    // to enable enum questiontype in template 
     QuestionType = QuestionType;
-    alpha: string[] = ["a", "b", "c", "d", "e", "..."];
+    alpha: string[] = ['a', 'b', 'c', 'd', 'e', '...'];
+
     constructor(private questionsService: QuestionsService, private dialog: MdDialog, private categoryService: CategoryService) {
-        this.getAllQuestions();
+    }
+
+    ngOnInit() {
+         this.getAllQuestions();
         this.getAllCategories();
     }
+
     isCorrectAnswer(isAnswer: boolean) {
         if (isAnswer) {
-            return "correct";
+            return 'correct';
         }
     }
-    //To Get All The categories
+
+    // to Get All The categories
     getAllCategories() {
         this.categoryService.getAllCategories().subscribe((CategoriesList) => {
             this.categoryArray = CategoriesList;
         });
     }
+
     getAllQuestions() {
         this.questionsService.getQuestions().subscribe((questionsList) => {
             this.questionDisplay = questionsList;
         });
     }
-    // Open Add Category Dialog
+
+    // open Add Category Dialog
     addCategoryDialog() {
         this.dialog.open(AddCategoryDialogComponent);
     }
-    // Open Delete Category Dialog
+
+    // open Rename Category Dialog
+    renameCategoryDialog(category: any) {
+        var renameDialogInstance = this.dialog.open(RenameCategoryDialogComponent).componentInstance;
+        renameDialogInstance.category = JSON.parse(JSON.stringify(category));
```

Comment:
A common use of JSON is to exchange data to/from a web server.When sending data to a web server, the data has to be a string, So Convert a Typescript object into a string with JSON.stringify().
so i used it